boa/src/environment/function_environment_record.rs

```diff
@@ -57,17 +59,43 @@ pub struct FunctionEnvironmentRecord {
 }
 
 impl FunctionEnvironmentRecord {
-    pub fn bind_this_value(&mut self, value: Value) -> Result<Value, ErrorKind> {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(
+        f: GcObject,
+        this: Option<Value>,
+        outer: Option<Environment>,
+        binding_status: BindingStatus,
+        new_target: Value,
+    ) -> Environment {
+        let mut func_env = FunctionEnvironmentRecord {
+            declarative_record: DeclarativeEnvironmentRecord {
+                env_rec: FxHashMap::default(),
+                outer_env: outer, // this will come from Environment set as a private property of F - https://tc39.es/ecma262/#sec-ecmascript-function-objects
+            },
+            function: f,
+            this_binding_status: binding_status,
+            home_object: Value::undefined(),
+            new_target,
+            this_value: Value::undefined(),
+        };
+        // If a `this` value has been passed, bind it to the environment
+        if let Some(v) = this {
+            func_env.bind_this_value(v).unwrap();
+        }
+        Gc::new(GcCell::new(Box::new(func_env)))
+    }
+
+    pub fn bind_this_value(&mut self, value: Value) -> Result<Value> {
         match self.this_binding_status {
             // You can not bind an arrow function, their `this` value comes from the lexical scope above
             BindingStatus::Lexical => {
                 panic!("Cannot bind to an arrow function!");
             }
             // You can not bind a function twice
-            BindingStatus::Initialized => Err(ErrorKind::new_reference_error(
-                "Cannot bind to an initialised function!",
-            )),
-
+            BindingStatus::Initialized => {
+                todo!();
+                // context.throw_reference_error("Cannot bind to an initialised function!")
+            }
```

Comment:
> Can you explain switching to `todo!()`?

It's because we would need to take `&mut Context` to throw, but we use it during the `Context` creation time in line 83 (I don't think this is spec compliant though, checking the spec it's not called there) Since this PR is not focused on making function calls/environments spec compliant I left it as `todo`